contracts/utils/IMultiRateLimited.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+/// @title interface for putting a rate limit on how fast a contract can perform minting
+/// @author Fei Protocol
+interface IMultiRateLimited {
+
+    /// @notice the rate per second for each minter
+    function individualRateLimitPerSecond(address) external view returns(uint256);
+
+    /// @notice the last time the buffer was used by each minter
+    function individualLastBufferUsedTime(address) external view returns(uint256);
+
+    /// @notice the cap of the buffer that can be used at once
+    function individualBufferCap(address) external view returns(uint256);
+
+    // ----------- Events -----------
+    
+    /// @notice emitted when a buffer is eaten into
+    event IndividualBufferUsed(address minter, uint256 amountUsed, uint256 bufferRemaining);
```

Comment:
`minter` references here should also be updated to `throttledAddress` or whichever variable name we use to reference that this contract applies more broadly